[Lin Weilin] iP

weilin1202 opened the pull request:
# BearyBear
>“Your mind is for having ideas, not holding them.” – David Allen [source](https://dansilvestre.com/productivity-quotes/)

BearyBear frees your mind of having to remember things you need to do. It's,

- text-based
- **easy** to learn
- *SUPER FAST* to use

## All you need to do is,

1. *download* it from [here](https://github.com/weilin1202/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

## Features:
- [x] Manage Different Types of Tasks
- [x] Update status of tasks
- [x] Save the tasks locally 


---------------------------------------------------------------------------------------------------------------------
If you Java programmer, you can use it to practice Java too. Here's the main method:
```ruby
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

weilin1202 committed:
add level 1

weilin1202 committed:
add level 2

weilin1202 committed:
Add level 3

weilin1202 committed:
Add level 4

weilin1202 committed:
Add UI test

weilin1202 committed:
Add level 5

weilin1202 committed:
Add level 6

weilin1202 committed:
Add level 7

weilin1202 committed:
Add level 8

nicleejy commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+
+public class Deadline extends Task{
+
+    String taskDescription;
+    LocalDate deadLine;
```
Since these class fields have corresponding getters and setters, it might be better to make them private to prevent any unintended modifications.

nicleejy commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,112 @@
+                System.out.println("Got it. I've added this task: \n  " + task +
+                        "\nNow you have " + taskList.size() + " tasks in the list.");
+
+            } else if(input.length() > 5 && (input.substring(0, 5)).equals("event")) {
```
Try to surround reserved words with a whitespace for consistency

nicleejy commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,30 @@
+import java.time.format.DateTimeFormatter;
+
+public class Event extends Task{
+    String taskDescription;
+    LocalDate startDate;
+    LocalDate endDate;
```
Same goes for this, regarding making class fields private

nicleejy commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,28 @@
+        return this.deadLine.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " ( by: " + this.getDeadline() + " )" ;
+    }
```
This is a good way to separate the different display behaviors of different task types 👍

nicleejy reviewed (commented):
Overall, the coding standard has been followed, just apart from some minor issues! 👍 In most of your classes, do remember to keep fields private to prevent unwanted modifications by the user. It might also be good to declare Task as an abstract class since it should not be instantiated at any point in the program. Do let me know if you have any queries regarding my comments :)

weilin1202 committed:
add A-MoreOOP and A-Packages

weilin1202 committed:
Merge branch 'add-gradle-support' of https://github.com/weilin1202/ip

weilin1202 committed:
Add A-JUnit

weilin1202 committed:
Add A-JavaDoc

weilin1202 committed:
Add A-CodingStandard

weilin1202 committed:
Add Level-9

weilin1202 committed:
Add Level-10

weilin1202 committed:
Add A-CheckStyle

weilin1202 committed:
Completed A-Jar

weilin1202 committed:
Implemented A-Jar using Gradle again

weilin1202 committed:
Update Jar file

weilin1202 committed:
Add A-Assertions

weilin1202 committed:
New message

weilin1202 committed:
Improve Code Quality an OOP

weilin1202 committed:
Merge pull request #2 from weilin1202/branch-A-Assertions

Add A-Assertions

weilin1202 committed:
Merge branch 'master' of https://github.com/weilin1202/ip

weilin1202 committed:
Merge pull request #3 from weilin1202/branch-A-CodeQuality

Parser performCommand: Code quality has not been achieved because method is too long and OOP is not fully achieved 

PerformCommand is supposed to handle the general processing of command, checking the performing comman shall be done in separate classes for each different cases. 

Created several classes in Parser to handle different types of commands to make the code quality better, ensuring methods are not too long and as well as to make full use of OOP.

weilin1202 committed:
Merge branch 'branch-A-CodeQuality'

weilin1202 committed:
Merge branch 'master' of https://github.com/weilin1202/ip

weilin1202 committed:
Storage class: create file and is stored locally with a specific customised path for my hard disk

The user using Duke might not have same file path in local hard disk

The file path has thus been changed to ./data/duke.txt which makes it more general than the previous iteration

Let's create a new text file in ./data/duke.txt instead of C:\Users\linwe\Documents\taskSaved.txt!

weilin1202 committed:
Classes that handle separate commands in Parser folder: The checking of the validity of inputs are not robust

Checking not being robust will result in Duke mis-intepereting an input as valid when it is not, thereby causes execption being thrown without catching

Included more robust checking to ensure the inputs are in correct format, before a task object is created

Let's check the inputs by trying throwing and catching exception when inputs are incorrect!

weilin1202 committed:
Cannot detect duplicates in the addition of tasks

Same tasks can be added multiple times

Add C-DetectDuplicates

It is more user friendly so that users would not accidentally add
same tasks into the task list.

Let's
* Check whether there are existing same task in task list as the task
that user has keyed in
* Include the equals function in task to check whether two tasks are
the same or not

weilin1202 committed:
Moved the printing of String from Parser class to Ui class as Ui settles
the interaction with User and not the parser

weilin1202 committed:
Resolved the bugs discovered in Storage class after testing
and improved implementation of the different tasks

weilin1202 committed:
Improved the existing GUI

weilin1202 committed:
Add A-Personality to Duke

weilin1202 committed:
Add the javadocs for newly implemented methods and classes
Improve code style and quality and passed checkstyle

weilin1202 committed:
Added assertions and improve code quality of storage

weilin1202 committed:
A-UserGuide implementation

weilin1202 committed:
Completed version 1.0

weilin1202 committed:
Updated user guide

weilin1202 committed:
Update User Guide

weilin1202 committed:
Updated Storage file

weilin1202 committed:
Resolved background issue

weilin1202 committed:
Update User Guide

weilin1202 committed:
Update user guide

weilin1202 committed:
Updated comment in Storage class

weilin1202 committed:
Updated User guide

weilin1202 committed:
Updated the date checking when handling deadline and event timeline
Entering dates before the current date is not allowed
For event, start date cannot be after end date

weilin1202 committed:
Updated User Guide

weilin1202 committed:
Update user guide

weilin1202 committed:
Improved implementation of checking input for adding of tasks.
Settle case when input is missing or is filled with spaces

weilin1202 committed:
Resolved checkstyle issues

weilin1202 committed:
Re-uploaded Ui and corrected an extra space in the exception message

weilin1202 committed:
Update User Guide
